File: src/main/java/com/cse3063f25grp1/intent/RuleIntentDetector.java
package com.cse3063f25grp1.intent;

import java.util.List;
import java.util.Map;
import com.cse3063f25grp1.model.Intent;

public class RuleIntentDetector implements IntentDetector {

    private final Map<Intent, List<String>> rules;
    private final List<Intent> priorityOrder;

    // Yeni constructor: priorityOrder ConfigLoader tarafından gönderilecek
    public RuleIntentDetector(Map<Intent, List<String>> rules,
            List<Intent> priorityOrder) {
        this.rules = rules;
        this.priorityOrder = priorityOrder;
    }

    @Override
    public Intent detect(String question) {
        if (question == null || question.trim().isEmpty()) {
            return Intent.Unknown;
        }

        String lower = question.toLowerCase();

        // INTENT PRIORITY SIRASINA GÖRE KONTROL
        for (Intent intent : priorityOrder) {
            List<String> keywords = rules.get(intent);
            if (keywords == null)
                continue;

            for (String keyword : keywords) {
                if (keyword != null && lower.contains(keyword.toLowerCase())) {
                    return intent;
                }
            }
        }

        return Intent.Unknown;
    }
}

File: iteration1_py/src/intent/rule_intent_detector.py
from typing import Dict, List

from src.model.intent import Intent
from src.intent.intent_detector import IntentDetector


class RuleIntentDetector(IntentDetector):
  

    def __init__(self, rules: Dict[Intent, List[str]], priority_order: List[Intent]) -> None:
        self.__rules: Dict[Intent, List[str]] = rules
        self.__priority_order: List[Intent] = priority_order
    
    def detect(self, question: str) -> Intent:
        if question is None or question.strip() == "":
            return Intent.UNKNOWN

        lower: str = question.lower()

        # Check intents in priority order (same logic as Java)
        for intent in self.__priority_order:
            keywords: List[str] = self.__rules.get(intent, [])

            for keyword in keywords:
                if keyword and keyword.lower() in lower:
                    return intent

        return Intent.UNKNOWN
